### ISSUE TITLE
[Bug]: Accessibility issue with rating component

### ISSUE THREAD

**Issue by bhanu345-cmd** [microsoft/fluentui#28820]
### Library

React / v8 (@fluentui/react)

### System Info

```shell
System:
    OS: Windows 10 10.0.22621
    CPU: (8) x64 Intel(R) Core(TM) i5-10210U CPU @ 1.60GHz
    Memory: 3.55 GB / 15.78 GB
  Browsers:
    Edge: Spartan (44.22621.963.0), Chromium (108.0.1462.54)
    Internet Explorer: 11.0.22621.1
```

### Are you reporting Accessibility issue?

None

### Reproduction

https://codesandbox.io/s/old-haze-tz98z6

## Actual Behavior
<button class="ms-Rating-button ratingButton-115 ms-Rating--small ratingStarIsSmall-116" id="Rating0-star-0" role="radio" type="button" aria-checked="false" tabindex="0"><span id="RatingLabel1-1" class="ms-Rating-labelText labelText-118"></span><div class="ms-RatingStar-container ratingStar-112"><i data-icon-name="FavoriteStar" aria-hidden="true" class="ms-RatingStar-back ratingStarBack-123"></i><i data-icon-name="FavoriteStar" aria-hidden="true" class="ms-RatingStar-front ratingStarFront-124" style="width: 0%;"></i></div></button>

Element does not have inner text that is visible to screen readers
aria-label attribute does not exist or is empty
aria-labelledby attribute does not exist, references elements that do not exist or references elements that are empty
Element has no title attribute
Element's default semantics were not overridden with role="none" or role="presentation"

<img width="445" alt="image" src="https://github.com/microsoft/fluentui/assets/63151401/413ae6b8-180b-447a-a4e8-a3c7d8aafe00">

Reproduce steps:
1)open code sandbox browser in new window
2)run accessibility insights for web. You we will able to see the accessibility issue with rating component

## Expected Behavior
Any one of the properties mentioned above should be present.

### Requested priority

Blocking

### Are you willing to submit a PR to fix?

no

### Validations

- [X] Check that there isn't already an issue that reports the same bug to avoid creating a duplicate.
- [X] The provided reproduction is a minimal reproducible example of the bug.
